### PR TITLE
Simplify state management in summonerd

### LINKED ISSUE #3221: Summonerd shared state refactor
One way to simplify state management which would help make a progress web view easier is to centralize the management of the queue into a single piece of locked state, rather than spreading it out via channels. This would allow the server, the coordinator, and the web view to all coordinate via accessing this state, which would greatly simplify the current code.

More concretely, we would have an `Arc<RWLock<ParticipantQueue>>` or something akin to that, which would then be accessed as needed by these different components.

### PR DESCRIPTION
**Pull request by cronokirby**
Closes #3221

### CODE REVIEW

**Comment by cronokirby** [comment]
https://github.com/penumbra-zone/penumbra/actions/runs/6635035943 the full summoning smoke test passed successfully as well, but a careful review is still warranted imo

File: tools/summonerd/src/coordinator.rs [hunk 1]
```diff
@@ -1,90 +1,69 @@
...
         .await
         {
             Ok(Ok(_)) => Ok(()),
             Err(_) => {
                 tracing::info!("STRIKE (timeout)");
-                self.storage.strike(&contributor).await?;
+                self.storage.strike(&address).await?;
                 Ok(())
             }
             Ok(Err(e)) => Err(e),
```
Comment (hdevalence) [inline tools/summonerd/src/coordinator.rs:56]: Why do we want to bubble up an error here if `contribute_inner` fails? If a participant does something bad (e.g., submitting something that fails validation), won't this still crash the server?

File: tools/summonerd/src/participant.rs [hunk 2]
```diff
@@ -11,14 +11,14 @@ use penumbra_proto::{
...
-use tokio::sync::mpsc;
+use tokio::sync::{mpsc, Mutex};
 use tonic::{Status, Streaming};
 
 use crate::phase::Phase;
 
 pub struct Participant {
     address: Address,
-    rx: Streaming<pb::ParticipateRequest>,
+    rx: Mutex<Streaming<pb::ParticipateRequest>>,
```
Comment (hdevalence) [inline tools/summonerd/src/participant.rs:21]: I don't think we should be using a `Mutex` here, why are we sharing access to the participant's messages?

**Comment by hdevalence** [comment]
The comments above are largely stylistic; I'm not sure that the new code structure is optimal, but I think it may be good enough for the purposes of our ceremony, so we could merge this and move forward.

**Pull request merged**